catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/CsexraAttribute.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.transformer.nitf.common;
+
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.codice.alliance.catalog.core.api.impl.types.IsrAttributes;
+import org.codice.alliance.catalog.core.api.types.Isr;
+import org.codice.imaging.nitf.core.tre.Tre;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.MetacardType;
+
+enum CsexraAttribute implements NitfAttribute<Tre> {
+
+    @SuppressWarnings("RedundantTypeArguments")
+    SNOW_DEPTH_MIN(Isr.SNOW_DEPTH_MIN_CENTIMETERS,
+            NitfConstants.SNOW_DEPTH_CAT,
+            getSnowDepthAccessorFunction(Pair<Float, Float>::getLeft),
+            new IsrAttributes()), @SuppressWarnings("RedundantTypeArguments")
+    SNOW_DEPTH_MAX(Isr.SNOW_DEPTH_MAX_CENTIMETERS,
+            NitfConstants.SNOW_DEPTH_CAT,
+            getSnowDepthAccessorFunction(Pair<Float, Float>::getRight),
+            new IsrAttributes()), PREDICTED_NIIRS(Isr.NATIONAL_IMAGERY_INTERPRETABILITY_RATING_SCALE,
+            NitfConstants.PREDICTED_NIIRS,
+            tre -> {
+                return Optional.ofNullable(TreUtility.getTreValue(tre,
+                        NitfConstants.PREDICTED_NIIRS))
+                        .filter(String.class::isInstance)
+                        .map(String.class::cast)
+                        .map(CsexraAttribute::parseNiirs)
+                        .orElse(null);
```

Comment:
I don't know if it's just me, but having to go through an `Optional` just to avoid an if statement seems a little over kill and makes the code much harder to understand.

The non-functional code doesn't look too bad to me and seems way easier to understand:

```
Serializable value = TreUtility.getTreValue(tre, NitfConstants.PREDICTED_NIIRS);

if (value instanceof String) {
    return parseNiirs((String) value);
}

return null;
```